velox/connectors/clp/search_lib/ir/ClpIrVectorLoader.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/clp/search_lib/ir/ClpIrVectorLoader.h"
+#include "velox/connectors/clp/search_lib/BaseClpCursor.h"
+
+namespace facebook::velox::connector::clp::search_lib {
+
+void ClpIrVectorLoader::loadInternal(
+    RowSet rows,
+    ValueHook* hook,
+    vector_size_t resultSize,
+    VectorPtr* result) {
+  auto vector = *result;
+  for (int vectorIndex : rows) {
+    auto& logEvent = filteredLogEvents_->at(vectorIndex);
+    // TODO: also need to support auto-generated keys
+    auto userGenNodeIdValueMap = logEvent->get_user_gen_node_id_value_pairs();
+    vector->setNull(vectorIndex, true);
+    if (0 == userGenNodeIdValueMap.count(nodeId_)) {
+      continue;
+    }
+    auto value = userGenNodeIdValueMap.at(nodeId_);
+    if (!value.has_value()) {
+      continue;
+    }
```

Comment:
```suggestion
    auto const value_it{userGenNodeIdValueMap.find(nodeId_)};
    if (userGenNodeIdValueMap.end() == value_it || false == value_it->has_value()) {
      continue;
    }
    auto const& value{*value_it};
```
Maybe something more like this? Avoids doing the lookup twice + makes the special case handling a bit easier to follow imo.